src/metaswitch/clearwater/config_manager/config_access.py

```diff
@@ -153,9 +151,12 @@ def get_config_and_index(self, config_type):
             log.debug("Reading etcd config from '%s", key_path)
             download = self._etcd_client.read(key_path)
         except etcd.EtcdKeyNotFound:
-            log.error("etcd key %s is not present in the database", key_path)
-            raise ConfigDownloadFailed(
-                CONFIG_TYPE_DOESNT_EXIST.format(config_type))
+            # If the key isn't present, create an empty file with a default
+            # revision number. Users can add the first revision!
+            log.info("etcd key %s is not present in the database", key_path)
+            first_value = ""
```

Comment:
As a user I would be confused about an empty file without any further prompt that this is expected. So maybe we could do something like setting `first_value = "# The etcd cluster didn't have any config stored yet, please delete this line and add the first revision."`? Or have a prompt to the user saying something to the same effect?

Reply:
I don't want to add a value because the different configuration files might have different commenting styles (and I don't think it's good form to give the user a file they can't upload). I have added a warning that will be displayed to the user to let them know what's going on.